libcodechecker/analyze/analyzers/analyzer_types.py

```diff
@@ -432,20 +419,9 @@ def construct_store_handler(buildaction,
     """
     Construct a result handler for storing results in a database.
     """
-    assert buildaction.analyzer_type in supported_analyzers, \
-        'Analyzer types should have been checked already.'
-
-    if buildaction.analyzer_type == CLANG_SA:
-        res_handler = result_handler_plist_to_db.PlistToDB(
-            buildaction,
-            None,
-            run_id)
-
-    elif buildaction.analyzer_type == CLANG_TIDY:
-        res_handler = result_handler_clang_tidy.ClangTidyPlistToDB(
```

Comment:
Yes the storing of plist files to the database should not depend on the analyzer which generated it.
This functionality should be refactored so that there should be only one plist to database handler.

I did not wanted to increase the size of this pull request with this refactoring. It should be a separate development.